test/distributed/checkpoint/test_state_dict.py

```diff
@@ -590,46 +592,73 @@ def test_non_persistent_buffers(self) -> None:
     @with_comms
     @skip_if_lt_x_gpu(2)
     def test_broadcast_from_rank0(self) -> None:
-        model = CompositeParamModel(device=torch.device("cuda"))
-        optim = torch.optim.Adam(model.parameters())
-        fsdp_model = FSDP2(copy.deepcopy(model))
-        fsdp_optim = torch.optim.Adam(fsdp_model.parameters())
+        def inner_test(wrapper):
+            model = CompositeParamModel(device=torch.device("cuda"))
+            optim = torch.optim.Adam(model.parameters())
+            fsdp_model = wrapper(copy.deepcopy(model))
+            fsdp_optim = torch.optim.Adam(fsdp_model.parameters())
 
-        batch = torch.rand(8, 100, device="cuda")
-        model(batch).sum().backward()
-        optim.step()
-        states, _ = get_state_dict(model, optim)
+            batch = torch.rand(8, 100, device="cuda")
+            model(batch).sum().backward()
+            optim.step()
+            states, optim_states = get_state_dict(model, optim)
 
-        fsdp_model(batch).sum().backward()
-        fsdp_optim.step()
+            fsdp_model(batch).sum().backward()
+            fsdp_optim.step()
 
-        def check(equal):
-            fsdp_states, fsdp_optim_states = get_state_dict(
-                fsdp_model, fsdp_optim, options=StateDictOptions(full_state_dict=True)
-            )
-            if equal:
-                self.assertEqual(states, fsdp_states)
+            def check(equal):
```

Comment:
do we need to check `get_optimizer_state_dict` as well? in torchtune, we call model and optimizer sd separately

Reply:
oh yes, some how that was removed during rebasing, will add it back.